sound/soc/sof/intel/hda-ipc.c

```diff
@@ -145,10 +145,6 @@ irqreturn_t hda_dsp_ipc_irq_thread(int irq, void *context)
 	u32 msg;
 	u32 msg_ext;
 
-	/* here we handle IPC interrupts only */
-	if (!(sdev->irq_status & HDA_DSP_ADSPIS_IPC))
-		return IRQ_NONE;
-
```

Comment:
well, no. If you remove this, then you have absolutely no need for irq_status, this makes no sense.

Reply:
> You made a change that just invalidates everything we've talked about. If you remove the check for the irq_status, then there is no possible race condition and the irq_status is useless.
> I can't merge stuff that makes no sense, sorry.

That's true, sorry for didn't think more about it, let me clean it up.